haproxy/haproxy.cfg

```
@@ -0,0 +1,46 @@
+# Setting `log` here with the address of 127.0.0.1 will have the effect
+# of haproxy sending the udp log messages to its own rsyslog instance
+# (which sits at `127.0.0.1`) at the `local0` facility including all
+# logs that have a priority greater or equal to the specified log level
+# log 127.0.0.1 local0 warning
+global
+  maxconn 4096
+  lua-load /usr/local/etc/haproxy/parse_basic.lua
+  lua-load /usr/local/etc/haproxy/parse_cookie.lua
+  lua-load /usr/local/etc/haproxy/replace_password.lua
+  log /dev/log len 65535 local2 info
+
+defaults
+  mode http
+  log global
+  option dontlognull
+  option http-ignore-probes
+  timeout client 150000
+  timeout server 3600000
+  timeout connect 15000
+  stats enable
+  stats refresh 30s
+  stats auth $COUCHDB_USER:$COUCHDB_PASSWORD
+  stats uri /haproxy?stats
+
+frontend http-in
+  bind  $HAPROXY_IP:$HAPROXY_PORT
+  acl has_user req.hdr(x-medic-user) -m found
+  acl has_cookie req.hdr(cookie) -m found
+  acl has_basic_auth req.hdr(authorization) -m found
+  declare capture request len 400000
+  http-request set-header x-medic-user %[lua.parseBasic] if has_basic_auth
+  http-request set-header x-medic-user %[lua.parseCookie] if !has_basic_auth !has_user has_cookie
+  http-request capture req.body id 0 # capture.req.hdr(0)
+  http-request capture req.hdr(x-medic-service) len 200 # capture.req.hdr(1)
+  http-request capture req.hdr(x-medic-user) len 200 # capture.req.hdr(2)
+  http-request capture req.hdr(user-agent) len 600 # capture.req.hdr(3)
+  capture response header Content-Length len 10 # capture.res.hdr(0)
+  log-format "%ci,%ST,%[capture.req.method],%[capture.req.uri],%[capture.req.hdr(1)],%[capture.req.hdr(2)],'%[capture.req.hdr(0),lua.replacePassword]',%B,%Tr,%[capture.res.hdr(0)],'%[capture.req.hdr(3)]'"
```

I found it very useful for debugging knowing which server responds, I added `%s` to get that: 
```suggestion
  log-format "%ci,%s,%ST,%[capture.req.method],%[capture.req.uri],%[capture.req.hdr(1)],%[capture.req.hdr(2)],'%[capture.req.hdr(0),lua.replacePassword]',%B,%Tr,%[capture.res.hdr(0)],'%[capture.req.hdr(3)]'"
```
Result looks like: 
```
May  5 07:04:29 haproxy haproxy[25]: 192.168.160.6,couchdb2,200,GET,/medic-test/settings,api,admin,'-',96924,3,96654,'node-fetch/1.0 (+https://github.com/bitinn/node-fetch)'
```